therapy/etl/drugsatfda.py

```diff
@@ -0,0 +1,157 @@
+        """
+        r = requests.get("https://api.fda.gov/download.json")
+        if r.status_code == 200:
+            json = r.json()
+            try:
+                date_raw = json["results"]["drug"]["drugsfda"]["export_date"]
+            except KeyError:
+                msg = "Unable to parse OpenFDA version API - check for breaking changes"
+                logger.error(msg)
+                raise DownloadException(msg)
+            return date_raw.replace("-", "")
+        else:
+            raise requests.HTTPError("Unable to retrieve version from FDA API")
+
+    def _load_meta(self) -> None:
+        """Add Drugs@FDA metadata."""
+        meta = {
+            "data_license": "CC0",
+            "data_license_url": "https://creativecommons.org/publicdomain/zero/1.0/legalcode",  # noqa: E501
+            "version": self._version,
+            "data_url": "https://open.fda.gov/apis/drug/drugsfda/download/",
+            "rdp_url": None,
+            "data_license_attributes": {
+                "non_commercial": False,
+                "share_alike": False,
+                "attribution": False,
+            }
+        }
+        assert SourceMeta(**meta)
+        meta["src_name"] = SourceName.DRUGSATFDA
+        self.database.metadata.put_item(Item=meta)
+
+    def _get_marketing_status(self, products: List, concept_id: str)\
+            -> Optional[str]:
+        """Get approval status from products list.
+        :param List products: list of individual FDA product objects
+        :param str concept_id: FDA application concept ID, used in reporting error
+            messages
+        :return: approval_status value if successful, None if ambiguous or unavailable
+        """
+        statuses_map = {
+            "Discontinued": ApprovalStatus.FDA_DISCONTINUED.value,
+            "Prescription": ApprovalStatus.FDA_PRESCRIPTION.value,
+            "Over-the-counter": ApprovalStatus.FDA_OTC.value,
+            "None (Tentative Approval)": ApprovalStatus.FDA_TENTATIVE.value,
+        }
+        statuses = [p["marketing_status"] for p in products]
+        if not all([s == statuses[0] for s in statuses]):
+            msg = (f"Application {concept_id} has inconsistent marketing "
+                   f"statuses: {statuses}")
+            logger.info(msg)
+            return None
+        else:
+            return statuses_map.get(statuses[0])
+
+    def _transform_data(self) -> None:
+        """Prepare source data for loading into DB."""
+        with open(self._src_file, "r") as f:
+            data = json.load(f)["results"]
+
+        for result in data:
+            if "products" not in result:
+                continue
+            products = result["products"]
+
+            app_no = result["application_number"]
+            if app_no.startswith("BLA"):
+                # ignore biologics license applications (tentative)
+                continue
+            concept_id = f"{NamespacePrefix.DRUGSATFDA.value}:{app_no}"
```

Comment:
I don't think we should be including the `ANDA` in the concept_id since searching `075036` in DrugsAtFDA website returns a result, but `ANDA075036` does not. If you think that the `ANDA` part is important, you could have a new `NamespacePrefix` like `drugsatfda.anda`

Reply:
Yeah, this is an interesting question, and it's sort of awkward because this naming scheme predates the notion of concept IDs and namespaces (and, like, the internet). Colloquially you typically refer to individual applications as "<category><number>" a la "ANDA093939", and that's how they're represented both within the raw data and in other sources (eg https://healthprovidersdata.com/hipaa/codes/NDC_0143-9505.aspx, http://www.hipaaspace.com/medical_billing/coding/national.drug.codes/txt/55390-112-99). But semantically it's clearly meant as a namespace subcategory, as reflected by the search engine's function. @mcannon068nw ?

Reply:
I could be wrong on this, but I think it makes sense to keep the ANDA part in some form, just because there are multiple types of drug applications (NDA, ANDA, BLA, etc.). I think we would be losing information if we drop the ANDA from ANDA075036.

Reply:
I went with sub-namespacing, a la `drugsatfda.anda:093939`. I think it's a bit more consistent with our naming practices and retains the relevant ANDA vs NDA distinction (plus possible BLA inclusion in the future).